src/overlay/OverlayManagerImpl.cpp

```diff
@@ -586,13 +629,24 @@ OverlayManagerImpl::tick()
         preferredToConnect =
             std::min(availablePendingSlots, preferredToConnect);
 
-        auto pendingUsedByPreferred =
+        pendingUsedByPreferred =
             connectTo(preferredToConnect, PeerType::PREFERRED);
 
         releaseAssert(pendingUsedByPreferred <= availablePendingSlots);
         availablePendingSlots -= pendingUsedByPreferred;
     }
 
+    // Only trigger reconnecting if:
+    //   * no outbound slots are available
+    //   * we didn't establish any new preferred peers connections (those
+    //      will evict regular peers anyway)
+    if (availableAuthenticatedSlots == 0 && pendingUsedByPreferred == 0)
```

Comment:
Should the second criteria just be that we can initiate connections? I don't see why trying to connect to a preferred peer should stop the node from triggering this logic

Reply:
I kept it like this because preferred peers have their own eviction policy when all connections are taken, so I didn't override it just in case (though it seems a bit arbitrary)

Reply:
what I mean is that if we have 2 pending slots available and 1 preferred peer to connect to, there is no reason to not trigger this logic.

Trying to connect to a preferred peer does not mean that we'll be able to connect to it, so in this case we'd just be wasting cycles. So I'd rather not couple those behaviors.